src/components/Account.tsx

```diff
@@ -21,25 +21,27 @@ import useBalance from '../hooks/useBalance';
 import AccountDisplay from './AccountDisplay';
 
 interface Props {
-  value: string;
   accountName?: string;
-  className?: string;
   chain?: string | undefined;
+  className?: string;
+  hideAddress?: boolean;
   isDerived?: boolean;
   onClick?: () => void;
+  value: string;
 }
 
-const Account = ({ accountName, value, chain, isDerived = false }: Props) => {
+const Account = ({ accountName, value, chain, hideAddress = false, isDerived = false }: Props) => {
   const { api, address } = useApiBalance(value, chain, isDerived);
 
   const state = useBalance(api, address, true);
 
   return (
     <AccountDisplay
-      accountName={accountName}
+      friendlyName={accountName}
       address={address}
       balance={state.formattedBalance}
-      isDerived={isDerived}
+      addressKind={isDerived ? 'companion' : undefined}
```

Comment:
seems weird to provide `undefined ` as a value, why not null ? or handle this logic in the component?

Reply:
Because `addressKind` is optional (so it can be `undefined`). Passing `null` would complicate the expected type and would introduce ambiguity between `null` and `undefined` - should the behavior be different? - as a component user you have no idea if it's going to be or not.

Reply:
We can't handle this logic in component, cause boolean flag `isDerived` is not sufficient to express different kinds the account may have (3 possibilities + string), unless you want to have `boolean | string | null` which IMHO is way more misleading.

Reply:
i missed the optional definition for `addressKind ` ... makes sense....